src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -0,0 +1,198 @@
+        option.label    = CharSpan::fromCharString(label);
+        option.mode     = mode;
+        option.modeTags = modeTags;
+        return option;
+    }
+
+    CHIP_ERROR Init();
+
+    // CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+
+    // Attribute setters
+    void UpdateStartUpMode(DataModel::Nullable<uint8_t> aNewStartUpMode);
+    void UpdateOnMode(DataModel::Nullable<uint8_t> aNewOnMode);
+    void UpdateCurrentMode(uint8_t aNewMode);
+
+    // Attribute getters
+    DataModel::Nullable<uint8_t> GetStartUpMode() const;
+    DataModel::Nullable<uint8_t> GetOnMode() const;
+    uint8_t GetCurrentMode() const;
+    EndpointId GetEndpointId() const { return mEndpointId; }
+
+    /**
+     * Returns true if the feature is supported.
+     * @param feature the feature to check.
+     */
+    bool HasFeature(Feature feature) const;
+
+    /**
+     * This function returns true if the mode value given matches one of the supported modes, otherwise it returns false.
+     * @param mode
+     */
+    bool IsSupportedMode(uint8_t mode);
+
+private:
+    EndpointId mEndpointId{};
+    ClusterId mClusterId{};
+
+    // Attribute data store
+    uint8_t mCurrentMode;
+    DataModel::Nullable<uint8_t> mStartUpMode;
+    DataModel::Nullable<uint8_t> mOnMode;
+
+    uint32_t mFeature;
+
+    /**
+     * This checks to see if this clusters instance is a valid ModeBase aliased cluster based on the AliasedClusters list.
+     * @return true if the clusterId of this instance is a valid ModeBase cluster.
+     */
+    bool isAliasCluster() const;
+
+    /**
+     * Internal change-to-mode command handler function.
+     */
+    void handleChangeToMode(HandlerContext & ctx, const Commands::ChangeToMode::DecodableType & req);
+
+public:
+    /**
+     * Creates a mode base cluster instance. The Init() function needs to be called for this instance to be registered and
+     * called by the interaction model at the appropriate times.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the ModeBase aliased cluster to be instantiated.
+     * @param aFeature The bitmask value that identifies which features are supported by this instance.
+     */
+    Instance(EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature) :
+        CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId)
+    {
+        mEndpointId = aEndpointId;
```

Comment:
This can be done via the static initializers list.

Reply:
done